build/go.mod

```diff
@@ -0,0 +1,33 @@
+module github.com/mattermost/mattermost-plugin-starter-template/build
```

Comment:
I added `build/go.mod` to lessen the scope of this PR a bit. This way we don't introduce a bunch of changes to use `mattermost/server/public` in this PR. That will be done in a separate effort to sync all plugin projects with starter template

Reply:
Good call :+1: